apps/server/src/modules/deletion/entity/deletion-request.entity.ts

```diff
@@ -0,0 +1,60 @@
+import { Entity, Index, Property } from '@mikro-orm/core';
+import { BaseEntityWithTimestamps } from '@shared/domain/entity/base.entity';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { EntityId } from '@shared/domain';
+import { DeletionDomainModel } from '../domain/types/deletion-domain-model.enum';
+import { DeletionStatusModel } from '../domain/types/deletion-status-model.enum';
+
+export interface DeletionRequestEntityProps {
+	id?: EntityId;
+	domain: DeletionDomainModel;
+	deleteAfter: Date;
+	itemId: EntityId;
+	status: DeletionStatusModel;
+	createdAt?: Date;
+	updatedAt?: Date;
+}
+
+@Entity({ tableName: 'deletionrequests' })
+@Index({ properties: ['_itemId', 'domain'] })
+export class DeletionRequestEntity extends BaseEntityWithTimestamps {
+	@Property()
+	deleteAfter: Date;
+
+	@Property()
+	_itemId: ObjectId;
+
+	get itemId(): EntityId {
+		return this._itemId.toHexString();
+	}
+
+	@Property()
+	domain: DeletionDomainModel;
```

Comment:
This name is fine, but if you'll change the name of the field that holds the reference to the "target" please change this one accordingly, e.g. to `targetRefDomain` to make it obvious it's bounded to the field that holds the id.

Reply:
done